src/main/java/com/elixr/poc/service/PurchaseService.java

```diff
@@ -20,17 +23,35 @@ public PurchaseService(PurchaseRepository purchaseRepository) {
      *
      * @param purchaseId
      * @return
-     * @throws NoRecordFoundException
+     * @throws IdNotFoundException
      */
-    public boolean deletePurchaseDetails(UUID purchaseId) throws NoRecordFoundException {
-        boolean success = false;
+    public boolean deletePurchaseDetails(UUID purchaseId) throws IdNotFoundException {
```

Comment:
You are already handling the IdNotFoundException. Why is it throwed again by the method?

Reply:
Removed